spyro/utils/utils.py

```diff
@@ -145,6 +146,204 @@ def communicate(array, my_ensemble):
+
+        self.active_boundaries = active_boundaries
+        if inverse_mask is False:
+            self._get_mask_conditional(Wave_obj)
+        elif inverse_mask is True:
+            self._get_mask_conditional_inverted(Wave_obj)
+        self.in_dg = dg
+        if dg is False:
+            self._get_mask_dofs(Wave_obj)
+        elif dg is True:
+            self._get_dg_mask(Wave_obj)
+
+    def _get_dg_mask(self, Wave_obj):
+        """
+        Calculates the DG mask.
+
+        Parameters:
+        - Wave_obj (object): The wave object containing the necessary data.
+
+        """
+        V_dg = FunctionSpace(Wave_obj.mesh, "DG", 0)
+        dg_mask = Function(V_dg)
+        dg_mask.interpolate(self.cond)
+        self.dg_mask = dg_mask
+
+    def _get_mask_conditional(self, Wave_obj):
+        """
+        Calculates the mask degrees of freedom based on the active boundaries.
+
+        Parameters:
+        - Wave_obj (object): The wave object containing the necessary data.
+
+        """
+        # Getting necessary data from wave object
+        active_boundaries = self.active_boundaries
+        self.z = Wave_obj.mesh_z
+        self.x = Wave_obj.mesh_x
+        if ("y_min" in active_boundaries) or ("y_max" in active_boundaries):
+            self.y = Wave_obj.mesh_y
+
+        # Getting mask conditional
+        first_boundary = True
+        for boundary in active_boundaries:
+            if first_boundary:
+                axis = boundary[0]
+                if boundary[-3:] == "min":
+                    cond = conditional(getattr(self, axis) < getattr(self, boundary), 1, 0)
+                elif boundary[-3:] == "max":
+                    cond = conditional(getattr(self, axis) > getattr(self, boundary), 1, 0)
+                else:
+                    raise ValueError(f"Boundary of {boundary} not possible")
+                first_boundary = False
+            else:
+                axis = boundary[0]
+                if boundary[-3:] == "min":
+                    cond = conditional(getattr(self, axis) < getattr(self, boundary), 1, cond)
+                elif boundary[-3:] == "max":
+                    cond = conditional(getattr(self, axis) > getattr(self, boundary), 1, cond)
+                else:
+                    raise ValueError(f"Boundary of {boundary} not possible")
+
+        self.cond = cond
+    
+    def _get_mask_conditional_inverted(self, Wave_obj):
+        """
+        Calculates the inverted mask degrees of freedom based on the active boundaries.
+
+        Parameters:
+        - Wave_obj (object): The wave object containing the necessary data.
+
+        """
+        # Getting necessary data from wave object
+        active_boundaries = self.active_boundaries
+        self.z = Wave_obj.mesh_z
+        self.x = Wave_obj.mesh_x
+        if ("y_min" in active_boundaries) or ("y_max" in active_boundaries):
+            self.y = Wave_obj.mesh_y
+
+        # Getting mask conditional
```

Comment:
Thanks for your suggestion. It simplifies the code. Tried code like this (but with cond assigned earlier), but it doesn't work because the int value is immutable. Uploaded a modified version of it using lists in the latest commit.